File: data/readable_format/fortran/file2245.f90
program test_main
  implicit none
  integer, parameter :: N = 100
  integer :: a(N,N,N), sa, result
  double precision :: b(N,N,N), sb
  real :: c(N,N,N), sc
  integer :: i, j, k

  sa = 0
  sb = 0.0d0
  sc = 0.0

  do k = 1, N
     do j = 1, N
        do i = 1, N
           a(i,j,k) = 1
        end do
     end do
  end do

  do k = 1, N
     do j = 1, N
        do i = 1, N
           b(i,j,k) = 0.5d0
        end do
     end do
  end do

  do k = 1, N
     do j = 1, N
        do i = 1, N
           c(i,j,k) = 0.25
        end do
     end do
  end do
  
  do k = 1, N
     do j = 1, N
        do i = 1, N
           sa = sa + a(i,j,k)
        end do
     end do
  end do

  do k = 1, N
     do j = 1, N
        do i = 1, N
           sb = sb + b(i,j,k)
        end do
     end do
  end do
  
  do k = 1, N
     do j = 1, N
        do i = 1, N
           sc = sc + c(i,j,k)
        end do
     end do
  end do
  
  result = 0
  if(sa /= N**3 .or. abs(sb-(N**3*0.5d0)) > 0.000001d0 .or. abs(sc-(N**3*0.25)) > 0.0001) then
     result = -1
  end if

  if(result == 0) then
     print *, "PASS"
  else
     print *, "ERROR"
  end if

end program test_main

File: data/readable_format/cpp/file2245.cpp
#include <iostream>
#include <cmath>
#include <vector>

const int N = 100;

int main() {
    std::vector<std::vector<std::vector<int>>> a(N, std::vector<std::vector<int>>(N, std::vector<int>(N, 1)));
    std::vector<std::vector<std::vector<double>>> b(N, std::vector<std::vector<double>>(N, std::vector<double>(N, 0.5)));
    std::vector<std::vector<std::vector<float>>> c(N, std::vector<std::vector<float>>(N, std::vector<float>(N, 0.25)));

    int sa = 0;
    double sb = 0.0;
    float sc = 0.0;

    for (int k = 0; k < N; ++k) {
        for (int j = 0; j < N; ++j) {
            for (int i = 0; i < N; ++i) {
                sa += a[i][j][k];
            }
        }
    }

    for (int k = 0; k < N; ++k) {
        for (int j = 0; j < N; ++j) {
            for (int i = 0; i < N; ++i) {
                sb += b[i][j][k];
            }
        }
    }

    for (int k = 0; k < N; ++k) {
        for (int j = 0; j < N; ++j) {
            for (int i = 0; i < N; ++i) {
                sc += c[i][j][k];
            }
        }
    }

    int result = 0;
    if (sa != N * N * N ||
        std::abs(sb - (static_cast<double>(N * N * N) * 0.5)) > 0.000001 ||
        std::abs(sc - (static_cast<float>(N * N * N) * 0.25)) > 0.0001) {
        result = -1;
    }

    if (result == 0) {
        std::cout << "PASS" << std::endl;
    } else {
        std::cout << "ERROR" << std::endl;
        return 1;
    }

    return 0;
}